### PR TITLE
TST: add asset_names to mock equities

### PR DESCRIPTION
**Pull request by jbredeche**

### CODE REVIEW

File: zipline/assets/synthetic.py [hunk 1]
```diff
@@ -60,7 +60,8 @@ def make_rotating_equity_info(num_assets,
 def make_simple_equity_info(sids,
                             start_date,
                             end_date,
-                            symbols=None):
+                            symbols=None,
+                            names=None):
```
Comment (richafrank) [inline zipline/assets/synthetic.py]: Would you add this to the docstring?

File: zipline/testing/fixtures.py [hunk 2]
```diff
@@ -346,6 +346,7 @@ class WithAssetFinder(WithDefaultDateBounds):
     """
     ASSET_FINDER_EQUITY_SIDS = ord('A'), ord('B'), ord('C')
     ASSET_FINDER_EQUITY_SYMBOLS = None
+    ASSET_FINDER_EQUITY_NAMES = None
```
Comment (richafrank) [inline zipline/testing/fixtures.py]: We should add this to the docstring as well.

File: zipline/assets/synthetic.py [hunk 3]
```diff
@@ -82,11 +83,16 @@ def make_simple_equity_info(sids,
...
+
+    if names is None:
+        names = map(lambda x: str(x) + " INC.", symbols)
+
     return pd.DataFrame(
         {
             'symbol': list(symbols),
             'start_date': pd.to_datetime([start_date] * num_assets),
             'end_date': pd.to_datetime([end_date] * num_assets),
+            'asset_name': list(names),
```
Comment (richafrank) [inline zipline/assets/synthetic.py]: This branch has failing python3 tests in the case where `symbols` is passed as a generator (actually the result of a call to `map`, which in python2 returns a `list`). The map is already consumed by the call to `list(symbols)`, so `names` is an empty list, which is a different length from the length of the list of symbols.

File: zipline/assets/synthetic.py [hunk 4]
```diff
@@ -82,11 +83,16 @@ def make_simple_equity_info(sids,
     num_assets = len(sids)
     if symbols is None:
         symbols = list(ascii_uppercase[:num_assets])
+
+    if names is None:
+        names = map(lambda x: str(x) + " INC.", symbols)
```
Comment (richafrank) [inline zipline/assets/synthetic.py]: My instinct is to use a list comprehension here, but definitely just a style choice.

File: zipline/testing/fixtures.py (hunk 2 above)
Comment (jbredeche) [inline zipline/testing/fixtures.py]: done

File: zipline/assets/synthetic.py (hunk 1 above)
Comment (jbredeche) [inline zipline/assets/synthetic.py]: done

File: zipline/assets/synthetic.py (hunk 4 above)
Comment (jbredeche) [inline zipline/assets/synthetic.py]: I'll switch it.  My Ruby instincts kicked in there.

File: zipline/assets/synthetic.py (hunk 3 above)
Comment (jbredeche) [inline zipline/assets/synthetic.py]: thanks.  fixed

File: zipline/assets/synthetic.py [hunk 5]
```diff
@@ -82,11 +87,18 @@ def make_simple_equity_info(sids,
...
         symbols = list(ascii_uppercase[:num_assets])
+    else:
+        symbols = list(symbols)
+
+    if names is None:
+        names = [str(s) + " INC." for s in symbols]
+
     return pd.DataFrame(
         {
             'symbol': list(symbols),
```
Comment (richafrank) [inline zipline/assets/synthetic.py]: nitpick - You don't need this extra call to `list` now. Not a big deal.

**Pull request merged**